onnxruntime/test/providers/cpu/math/det_op_test.cc

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+
+namespace onnxruntime {
+namespace test {
+
+TEST(DetOpTest, 2d) {
+  OpTester test("Det", 11);
+  test.AddInput<float>("X", {2, 2}, {0., 1., 2., 3.});
+  test.AddOutput<float>("Y", {}, {-2.});
+  test.Run();
+}
+
+TEST(DetOpTest, 2dWithBatchSize3) {
+  OpTester test("Det", 11);
+  test.AddInput<float>("X", {3, 2, 2}, {1., 2., 3., 4., 1., 2., 2., 1., 1., 3., 3., 1.});
```

Comment:
Related to above, consider adding a test case with input rank > 3.

Reply:
The first dimension would be the size of the batch. The second and third are the matrix dims. Under what conditions would the input rank be > 3?

Reply:
e.g. input x of shape `[2,3,4,5,5]` -> output y of shape `[2,3,4]`.